.github/workflows/github-actions.yml

```diff
@@ -4,30 +4,66 @@ on:
   push:
     paths-ignore:
       - 'README.md'
-      - 'Dockerfile'
-      - '.github/**'
       - '.dockerignore'
       - 'docker-compose.yml'
       - 'ansible/**'
   pull_request:
-    paths-ignore:
-      - 'README.md'
-      - 'Dockerfile'
-      - '.github/**'
-      - '.dockerignore'
-      - 'docker-compose.yml'
-      - 'ansible/**'
+    branches:
+      -  main
 
-jobs: 
+jobs:
   build:
     name: Build and test
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-      name: Checkout the code
 
-    - run: docker build --target ci-test --tag todo-app:ci-test .
-      name: Build the CI test image
+    - name: Build the CI test image
+      run: docker build --target ci-test --tag todo-app:ci-test .
+
+    - name: Run the CI test image
+      run: docker run --env-file ./.env_test todo-app:ci-test
+      
+  publish:
+    name: Push image to Docker Hub
+    needs: build
+    runs-on: ubuntu-latest
+    if: ${{ (github.event_name == 'push') && (github.ref == 'refs/heads/main') }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
```

Comment:
This looks nice - have you tried adding it to the test stage too, to see if it can reduce the length of this job? Not an important thing, but could be nice to try out

Reply:
Added. Thanks!